tests/sentry/api/endpoints/test_organization_releases.py

```diff
@@ -417,13 +417,88 @@ def test_release_stage_filter(self):
         )
+
         assert [r["version"] for r in response.data] == [not_adopted_release.version]
 
-        # TODO: Test release stage sort here. Not currently supported
-        # response = self.get_valid_response(
-        #     self.organization.slug, query=f"{RELEASE_STAGE_ALIAS}:[adopted,not_adopted,replaced]", sort="adopted"
-        # )
-        # assert [r["version"] for r in response.data] == [adopted_release.version, replaced_release.version, not_adopted_release.version]
+        response = self.get_valid_response(
+            self.organization.slug,
+            sort="adoption",
+        )
+        assert [r["version"] for r in response.data] == [
+            adopted_release.version,
+            replaced_release.version,
+            not_adopted_release.version,
+        ]
+        adopted_rpe.update(adopted=timezone.now() - timedelta(minutes=15))
+
+        # Replaced should come first now.
+        response = self.get_valid_response(
+            self.organization.slug,
+            sort="adoption",
+        )
+        assert [r["version"] for r in response.data] == [
+            replaced_release.version,
+            adopted_release.version,
+            not_adopted_release.version,
+        ]
+
+        response = self.get_valid_response(self.organization.slug, sort="adoption", per_page=1)
+        assert [r["version"] for r in response.data] == [
+            replaced_release.version,
+        ]
+        next_cursor = self.get_cursor_headers(response)[1]
+        response = self.get_valid_response(
+            self.organization.slug,
+            sort="adoption",
+            per_page=1,
+            cursor=next_cursor,
+        )
+        assert [r["version"] for r in response.data] == [
+            adopted_release.version,
+        ]
+        next_cursor = self.get_cursor_headers(response)[1]
+        response = self.get_valid_response(
+            self.organization.slug,
+            sort="adoption",
+            per_page=1,
+            cursor=next_cursor,
+        )
+        prev_cursor = self.get_cursor_headers(response)[0]
+        assert [r["version"] for r in response.data] == [
+            not_adopted_release.version,
+        ]
+        response = self.get_valid_response(
+            self.organization.slug,
+            sort="adoption",
+            per_page=1,
+            cursor=prev_cursor,
+        )
+        prev_cursor = self.get_cursor_headers(response)[0]
+        assert [r["version"] for r in response.data] == [
+            adopted_release.version,
+        ]
+        response = self.get_valid_response(
+            self.organization.slug,
+            sort="adoption",
+            per_page=1,
+            cursor=prev_cursor,
+        )
+        prev_cursor = self.get_cursor_headers(response)[0]
+        assert [r["version"] for r in response.data] == [
+            replaced_release.version,
+        ]
```

Comment:
It's probably fine to just test the next page cursor once, and maybe also testing going back a page once. I don't think we need to iterate through the whole test dataset though.

Reply:
....I wanted to be sure I wasn't getting lucky 😛